File: TourismAgency/AgencyServices/src/main/java/services/IAgencyService.java
package services;

import domain.Reservation;
import domain.TravelAgent;
import domain.Trip;
import domain.validators.ValidationException;
import repository.RepoException;

import java.time.LocalTime;

public interface IAgencyService {
     void login(TravelAgent user,IAgencyObserver client) throws ServiceException;
     Iterable<Trip> getAllTrips()throws ServiceException;
     Iterable<Trip> findTripsByNameAndHours(String name, LocalTime minHour, LocalTime maxHour)throws ServiceException;
     void addUser(TravelAgent newUser) throws ValidationException, RepoException,ServiceException;
     void addReservation(Reservation reservation) throws ValidationException, RepoException, ServiceException;
     void logout(TravelAgent user,IAgencyObserver client)throws ServiceException;
}

File: AgentieTurism/services/IAgencyService.cs
using System;
using System.Collections.Generic;
using model.domain;
namespace services
{
    public interface IAgencyService
    {
        void Login(TravelAgent user, IAgencyObserver client);
        IEnumerable<Trip> GetAllTrips();
        IEnumerable<Trip> FindTripsByNameAndHours(String name, int minHout, int maxHour);
        void AddUser(TravelAgent agent);
        void AddReservation(Reservation reservation);
        void Logout(TravelAgent user, IAgencyObserver client);
    }
}
